gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/linker/JmsDestinationLinker.java

```diff
@@ -0,0 +1,127 @@
+        // JNDI password
+        if (entity.getJndiPassword() != null) {
+            if (entity.getJndiPassword().startsWith("$L7C2$")) {
+                // Ignore passwords that are L7C2 encoded. We can't decode them anyways.
+                entity.setJndiPassword(null);
+            } else {
+                String storedPasswordReference = extractVariableName(entity.getJndiPassword());
+                if (!isEmpty(storedPasswordReference)) {
+                    Matcher matcher = secpassPattern.matcher(storedPasswordReference);
+                    if (matcher.matches()) {
+                        // the middle string between secpass and plaintext will be the stored password name
+                        String storedPasswordName = matcher.group(1);
+                        StoredPassword storedPassword = this.findStoredPasswordRef(entity, bundle, storedPasswordName);
+
+                        // replace the expression with the password reference by name
+                        entity.setJndiPasswordRef(storedPassword.getName());
+                        entity.setJndiPassword(null);
+                    }
+                }
+            }
+        }
+    }
+    
+    private void linkDestinationStoredPassword(JmsDestination entity, Bundle bundle) {
+        // Destination password
+        if (entity.getDestinationPassword() != null) {
+            if (entity.getDestinationPassword().startsWith("$L7C2$")) {
+                // Ignore passwords that are L7C2 encoded. We can't decode them anyways.
+                entity.setDestinationPassword(null);
+            } else {
+                String storedPasswordReference = extractVariableName(entity.getDestinationPassword());
+                if (!isEmpty(storedPasswordReference)) {
+                    Matcher matcher = secpassPattern.matcher(storedPasswordReference);
+                    if (matcher.matches()) {
+                        // the middle string between secpass and plaintext will be the stored password name
+                        String storedPasswordName = matcher.group(1);
+                        StoredPassword storedPassword = this.findStoredPasswordRef(entity, bundle, storedPasswordName);
+
+                        // replace the expression with the password reference by name
+                        entity.setDestinationPasswordRef(storedPassword.getName());
+                        entity.setDestinationPassword(null);
+                    }
+                }
+            }
+        }
+    }
+    
+    private void linkInboundAssociatedService(JmsDestination entity, Bundle bundle) {
+        if (entity.getInboundDetail() == null || 
+                entity.getInboundDetail().getServiceResolutionSettings() == null ||
+                entity.getInboundDetail().getServiceResolutionSettings().getServiceRef() == null ||
+                entity.getInboundDetail().getServiceResolutionSettings().getServiceRef().isEmpty() ) {
```

Comment:
Could we reduce this conditional block instead of a chaining null checks

Reply:
Done.